torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1651,3 +1654,33 @@ def test_function_not_on_callee(self):
                 AttributeError, "RPC pickler does not serialize"
             ):
                 rpc.rpc_sync(callee_worker, foo_add, args=())
+
+    def _create_rref(self):
+        owner_rank = (self.rank + 2) % self.world_size
+        return rpc.remote(
+            "worker{}".format(owner_rank),
+            torch.add,
+            args=(torch.zeros(2, 2), 1)
+        )
+
+    @dist_init
+    def test_user_rrefs_confirmed(self):
+        dst_rank = (self.rank + 1) % self.world_size
+        rref = self._create_rref()
+        ret = rpc.rpc_sync(
+            "worker{}".format(dst_rank),
+            check_rref_confirmed,
+            args=(rref,)
+        )
+        self.assertEqual(ret, True)
+
+    @dist_init
+    def test_user_rrefs_confirmed_remote(self):
```

Comment:
Is it possible to add a test case where a UserRRef is never confirmed and verify the appropriate behavior there? (i.e. stall forever and not run the function, throw an appropriate exception, etc).

Reply:
I thought about this. Looks like we would need to deplete the threads on the user side or owner side to prevent them either from sending out the RRef request or confirmation. But that threading behavior will change when we introduce async user function. Let me spend some more time to see if it is possible to test it without messing up with threads.